composer/algorithms/alibi/README.md

```diff
@@ -0,0 +1,108 @@
+# ⚖️ ALiBi
+
+[\[How to Use\]](#how-to-use) - [\[Suggested Hyperparameters\]](#suggested-hyperparameters) - [\[Technical Details\]](#technical-details) - [\[Attribution\]](#attribution)
+
+ `Natural Language Processing`
+
+ALiBi (Attention with Linear Biases) dispenses with position embeddings for tokens in transformer-based NLP models, instead encoding position information by biasing the query-key attention scores proportionally to each token pair’s distance. ALiBi yields excellent extrapolation to unseen sequence lengths compared to other position embedding schemes. We leverage this extrapolation capability by training with shorter sequence lengths, which reduces the memory and computation load.
+
+| ![Alibi](https://storage.googleapis.com/docs.mosaicml.com/images/methods/alibi.png) |
+|:--:
+|*The matrix on the left depicts the attention score for each key-query token pair. The matrix on the right depicts the distance between each query-key token pair. m is a head-specific scalar that is fixed during training. Figure from [Press et al., 2021](https://openreview.net/forum?id=R8sQPpGCv0).*|
+
+## How to Use
+
+### Functional Interface
+
+```python
+import torch
+import composer.functional as cf
+
+from composer.algorithms.alibi.gpt2_alibi import _attn
+from composer.algorithms.alibi.gpt2_alibi import enlarge_mask
+from transformers.models.gpt2.modeling_gpt2 import GPT2Attention
+
+def training_loop(model, train_loader):
+    cf.apply_alibi(model=model,
+                   heads_per_layer=12,
+                   max_sequence_length=256,
+                   position_embedding_attribute="module.transformer.wpe",
+                   attention_module=GPT2Attention,
+                   attr_to_replace="_attn",
+                   alibi_attention=_attn,
+                   mask_replacement_function=enlarge_mask)
+
+    opt = torch.optim.Adam(model.parameters())
+    loss_fn = F.cross_entropy
+    model.train()
+  
+    for epoch in range(num_epochs):
+        for X, y in train_loader:
+            y_hat = model(X)
+            loss = loss_fn(y_hat, y)
+            loss.backward()
+            opt.step()
+            opt.zero_grad()
+```
+
+### Composer Trainer
+
+```python
```

Comment:
@ajaysaini725 - Please add a bit of text at the beginning explaining what's happening.